File: wac3p2.py
import sys

cases = int(sys.stdin.readline())

for i in range(cases):
	total = int(sys.stdin.readline())
	not_cube = total / 3
	not_cube_int = total // 3

	modX = modY = 0
	if not_cube > not_cube_int + 0.5:
		modX = 1
	if not_cube != not_cube_int:
		modY = 1
	
	print((not_cube_int + modX) * (not_cube_int + modY) * not_cube_int)

File: wac3p2.cpp
#include <iostream>

int main() {
	std::cin.sync_with_stdio(0);
	std::cin.tie(0);
	
	int cases, total;
	int a, b, c;
	std::cin >> cases;
	for (int i = 0; i < cases; i++) {
		std::cin >> total;
		a = total / 3;
		b = (total - a) / 2;
		c = total - a - b;
		std::cout << (long long) a * b * c << '\n';
	}
	return 0;
}